sdk/src/settings/builder.rs

```diff
@@ -58,23 +60,36 @@ pub(crate) enum ThumbnailQuality {
 }
 
 /// Settings for controlling automatic thumbnail generation.
+#[cfg_attr(feature = "json_schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
-pub(crate) struct ThumbnailSettings {
+pub struct ThumbnailSettings {
     /// Whether or not to automatically generate thumbnails.
+    ///
+    /// The default value is true.
+    ///
+    /// <div class="warning">
+    /// This setting is only applicable if the crate is compiled with the `add_thumbnails` feature.
+    /// </div>
     pub enabled: bool,
     /// Whether to ignore thumbnail generation errors.
     ///
     /// This may occur, for instance, if the thumbnail media type or color layout isn't
     /// supported.
+    ///
+    /// The default value is true.
     pub ignore_errors: bool,
     /// The size of the longest edge of the thumbnail.
     ///
     /// This function will resize the input to preserve aspect ratio.
+    ///
+    /// The default value is 1024.
```

Comment:
I think we should change the default to 512 or something. We don't really need 1024.